src/utils/composables/useFieldValidation.js

```diff
@@ -0,0 +1,132 @@
+import { computed, onMounted, ref, toRef, watch } from 'vue'
+
+export function propsValidator4ValidateFunc(value, _props) {
+    if (value === null) {
+        return true
+    }
+    const returnObject = value('')
+    if (typeof returnObject !== 'object') {
+        return false
+    }
+    if ((!'valid') in returnObject) {
+        return false
+    }
+    if ((!'invalidMessage') in returnObject) {
+        return false
+    }
+    if (typeof returnObject.valid !== 'boolean') {
+        return false
+    }
+    if (
+        typeof returnObject.invalidMessage !== 'string' &&
+        !(returnObject.invalidMessage instanceof 'String')
+    ) {
+        return false
+    }
+    return true
+}
+
+export function useFieldValidation(
+    props,
+    model,
```

Comment:
I don't know if I like that you send the whole `props` Object here, as normally you should send refs or getters to a composable
https://vuejs.org/guide/reusability/composables#conventions-and-best-practices
and use `toValue` to make sure you get the value if it is a ref you got

The less `ref(...)` you declare here, the better it will react to changes in the inputs

Reply:
As discussed we leave it but I documented it